Dalamud/Game/Command/CommandManager.cs

```diff
@@ -135,66 +184,88 @@ public bool AddHandler(string command, CommandInfo info, string loaderAssemblyNa
     /// <inheritdoc/>
     public bool AddHandler(string command, CommandInfo info)
     {
-        if (info == null)
-            throw new ArgumentNullException(nameof(info), "Command handler is null.");
-
-        if (!this.commandMap.TryAdd(command, info))
-        {
-            Log.Error("Command {CommandName} is already registered.", command);
-            return false;
-        }
-
-        return true;
+        return this.AddHandler(command, info, null);
     }
 
     /// <inheritdoc/>
-    public bool RemoveHandler(string command)
+    [Experimental("Dalamud001")]
+    public bool AddCommand(string commandName, string helpMessage, Delegate func, bool showInHelp = true, int displayOrder = -1)
```

Comment:
Honestly not sure here re a bare `Delegate`. Maybe a new class like a `CommandExecutionContext` would be better? Would let us pass `args` as a bare string, "parsed" args, and other context info (e.g. if we're in a macro).